src/integrationTest/java/uk/gov/hmcts/reform/cpo/controllers/CasePaymentOrdersControllerAuthChecksIT.java

```diff
@@ -0,0 +1,237 @@
+package uk.gov.hmcts.reform.cpo.controllers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.cpo.data.CasePaymentOrderEntity;
+import uk.gov.hmcts.reform.cpo.payload.CreateCasePaymentOrderRequest;
+import uk.gov.hmcts.reform.cpo.payload.UpdateCasePaymentOrderRequest;
+import uk.gov.hmcts.reform.cpo.repository.CasePaymentOrdersAuditJpaRepository;
+import uk.gov.hmcts.reform.cpo.repository.CasePaymentOrdersJpaRepository;
+import uk.gov.hmcts.reform.cpo.utils.CasePaymentOrderEntityGenerator;
+
+import java.util.stream.Stream;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static uk.gov.hmcts.reform.cpo.controllers.CasePaymentOrdersController.CASE_PAYMENT_ORDERS_PATH;
+import static uk.gov.hmcts.reform.cpo.controllers.CasePaymentOrdersController.IDS;
+
+class CasePaymentOrdersControllerAuthChecksIT extends BaseMvcAuthChecks implements BaseMvcAuthChecks.AuthChecks {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private CasePaymentOrdersAuditJpaRepository casePaymentOrdersAuditJpaRepository;
+
+    @Autowired
+    private CasePaymentOrdersJpaRepository casePaymentOrdersJpaRepository;
+
+    @BeforeEach
+    @Transactional
+    void setUp() {
+        casePaymentOrdersJpaRepository.deleteAllInBatch();
```

Comment:
Is @Mockbean on service reduces the scope a bit ? (since full happy paths are covered in another IT tests already). Or do you wanna run this in full e2e setup.

Reply:
I was happiest leaving the full e2e as it ensures the happyPathRequestBuilder isn't broken in some other way.